.buildkite/pipelines/catalog/backstage_catalog_sync.yml

```diff
@@ -0,0 +1,25 @@
+# ==============================================================
+# Backstage Catalog Sync Pipeline
+# ==============================================================
+# Pipeline: Generate Backstage entries for Kibana packages in elastic/kibana-package-catalog
+# Description: Scans Kibana packages and generates Backstage component YAMLs in kibana-package-catalog repo.
+# Trigger: when PR merged to main branch
+# ==============================================================
+#
+
+steps:
+  - label: ':gear: Generate catalog components from Kibana packages'
+    key: generate_components
+    commands:
+      - set -euo pipefail
+      - git clone https://x-access-token:${GITHUB_TOKEN}@github.com/elastic/kibana-package-catalog.git ../kibana-package-catalog # clone kibana-package-catalog outside of kibana folder
```

Comment:
not sure this will work right away, I think we have to enable each repo we want to access on the github token somewhere.
Also, it looks a bit unsafe to have the token in the command, wouldn't it get printed as the name of the command executed?

Reply:
The previous implementation used github workflows for this and that worked. I refactored to Buildkite per request. Is there a Github token already set up on CI that you know of? If not, do we know how to get one?

Reply:
I see `https://github.com/elastic/kibana/blob/main/.buildkite/scripts/common/setup_job_env.sh#L18`, use that?

Reply:
yeah, maybe it's redacted 🤷 I never tested :) 

Ah, sorry, I misunderstood, this is our kibanamachine token, and not the github token that's provisioned with the agent. In this case, Kibanamachine should be able to R/W the repository

Reply:
Ok. Is this now accepted or are you still looking for a change?